htdocs/web_portal/controllers/service_group/add_service_group.php

```diff
@@ -67,16 +69,30 @@ function submit($user) {
     }
 }
 
-function draw($user) {
+function draw($user)
+{
     try {
-        // If the user is registered they're allowed to add a service group
+        $hasRolesAssigned = false;
+        $isUserValid = false;
+        $hasAdminCredentials = $user->isAdmin();
+
         if (is_null($user)) {
             throw new \Exception("Unregistered users can't create service groups.");
         }
 
+        $roleService = \Factory::getRoleService();
+        $hasRolesAssigned = $roleService->getUserRoles($user);
+
+        $isUserValid = $hasAdminCredentials ? true : !empty($hasRolesAssigned);
+
+        if (!$isUserValid) {
+            throw new Exception("You will need to have at least one role"
+                    . " assigned for adding a new service group.");
```

Comment:
```suggestion
            throw new Exception("You do not have permission to add a new"
                    . " ServiceGroup. To add a new ServiceGroup you require"
                    . " a role over an existing GOCDB enitity.");
```

Reply:
Sure, Greg. Updated the exception message with you suggestion.